pkg/oc/cli/cmd/debug.go

```diff
@@ -317,27 +306,80 @@ func (o *DebugOptions) Complete(cmd *cobra.Command, f kcmdutil.Factory, args []s
 		return err
 	}
 	o.ImageClient = imageClient.Image()
+
 	return nil
 }
+
 func (o DebugOptions) Validate() error {
+	if (o.AsRoot || o.AsNonRoot) && o.AsUser > 0 {
+		return fmt.Errorf("you may not specify --as-root and --as-user=%d at the same time", o.AsUser)
+	}
+	return nil
+}
+
+// Debug creates and runs a debugging pod.
+func (o *DebugOptions) RunDebug() error {
+	b := o.Builder().
+		WithScheme(ocscheme.ReadingInternalScheme, ocscheme.ReadingInternalScheme.PrioritizedVersionsAllGroups()...).
+		NamespaceParam(o.Namespace).DefaultNamespace().
+		SingleResourceType().
+		ResourceNames("pods", o.Resources...).
+		Flatten()
+	if len(o.FilenameOptions.Filenames) > 0 {
+		b.FilenameParam(o.ExplicitNamespace, &o.FilenameOptions)
+	}
+	one := false
+	infos, err := b.Do().IntoSingleItemImplied(&one).Infos()
+	if err != nil {
+		return err
+	}
+	if !one {
+		return fmt.Errorf("you must identify a resource with a pod template to debug")
```

Comment:
s/identify/specify/

Reply:
This can be fixed in followup, I didn't change the original strings.